src/encoded/types/file.py

```diff
@@ -473,6 +482,40 @@ def set_status(self, new_status, request, parent=True):
                     raise e
         return True
 
+    def _file_in_correct_bucket(self, request):
+        '''
+        Returns : boolean, current_path, destination_path
+        '''
+        return_flag = True
+        public_bucket = request.registry.settings.get('pds_public_bucket')
+        private_bucket = request.registry.settings.get('pds_private_bucket')
+        properties = self.upgrade_properties()
+        try:
+            external = self._get_external_sheet()
+        except HTTPNotFound:
+            # File object doesn't exist, leave it alone.
+            return (return_flag, None, None)
+        current_bucket = external.get('bucket')
+        current_key = external.get('key')
+        base_uri = 's3://{}/{}'
+        current_path = base_uri.format(current_bucket, current_key)
+        file_status = properties.get('status')
+        # Released restricted files should be in private bucket.
+        if file_status in self.private_s3_statuses or not self._should_set_object_acl():
+            if current_bucket == private_bucket:
```

Comment:
the return flag_is already true so we should be able to just
```
if not current_bucket == private_bucket:
     return_flag = False
```
Same for the public block.  Save a few lines of code